### ISSUE TITLE
Command Bytes for ep2231 lattego

### ISSUE THREAD

**Issue by Endorian88** [chris7topher/SmartPhilips2200#9]
hi, first of all thanks for this great project.

I have the following problem. The 2231 has the cappuccino button at the point where you have the steam button. However, when I try to make a cappuccino, the machine wants to make steam instead of Cappuccino. Looks funny but no Cappuccino for me. 
I also tried the bytes from the 3200, there is cappuccino then hot water and the latte only makes steam.

does anyone have the byte for cappuccino in the 2231? or maybe you have already upgraded your coffee machine?

Unfortunately I didn't know where else to ask.

**Comment by micki88** [comment]
Hi you need to read out your commands from the serial signal when button pressed on display.

**Comment by DivanX10** [comment]
Decrypting bytes in your case will not be difficult if you show patience, perseverance and a desire to find them. I will say more that it will be much easier to decrypt the bytes of your model than for the Philips 5400. At first it may seem that it is impossible to do this, but it is possible. Read how to decrypt bytes [here](https://community.home-assistant.io/t/esp-smart-offee-machine-philips-5400-series/568545/46?u=divanx10) and try to find them yourself. It would be great if we would make a similar recipe decoding table for the 2000/3000 models to make life easier for others

When deciphering recipes for Philips 5400, I did not immediately create such a table. First, I stuffed all the bytes I found, then, when I noticed the differences and the pattern, the picture began to clear up. I began to highlight what became clear and thus the overall picture is formed. Try to create a similar table by making different coffee drinks over and over again. It took me 1 kg of coffee beans to experiment, and instead of milk, I used water to save money. In order not to run around with a cup, I poured coffee into a saucepan and then poured it into the toilet, and so I was able to decipher the recipe for coffee drinks.

![image](https://github.com/chris7topher/SmartPhilips2200/assets/64090632/36810fe7-37c4-4f60-8e78-7e5b2a573ed5)

**Comment by flashmandv** [comment]
@DivanX10 can you help me with some ideas why my Phillips 3200 series is randomly starting the brewing process, when brew is selected, but before start_cmd is sent? 
I've read all info in github (4-5 repos). I also read your struggle with the 5400 series (**amazing job btw. Huge respect**!)

This is the code in the loop method that routes the machine Tx pin and it works, but random start cmds appear after I select (from ESP code or via machine button) the brew type (coffee, hot water..etc)
`while (swSer.available() > 0) {
      Serial.write(swSer.read());
    }`

I use same pins as Christopher code but these random start commands are ruining it all.

I commented almost all of the code to find the reason and there is nothing else to comment. This while that sends from SoftwareSerial to the Serial should be all fine..but no idea

Could it be my wiring schematics? 
<img width="1064" alt="image" src="https://github.com/chris7topher/SmartPhilips2200/assets/2466808/9efaaa70-ddb3-4f66-9844-e9adac0019f2">

Cable1: 5v- goes to the ESP and to coffee machine
Cable2: GND - routed through the transistor as per Christopher wiring
Cable3: GND2 - routed through the transistor (disconnecting it or wiring it directly does not turn on the display when I power it on from the ESP)
Cable4: Connected to the display (as it was before I touched the coffee machine)
Cable5: Rx - Goes to both display and ESP as per Christopher wiring
Cable6: Tx - routed through the ESP as per Christopher wiring
Cable7: Goes to the display as it was before I touch anything
Cable8: Goes to the display as it was before I touch anything

I may try to cut the 7 and 7 cable pins but doubt that it would work

Thank you!
I hope for your reply as I'm getting crazy

**Comment by micki88** [comment]
Are you usinig nodemcu? Probably noise... try switching hw and sw serial as pass thru with sw serial instead hw serial.

**Comment by flashmandv** [comment]
Yes, NodeMCU v2. I tried using SoftwareSerial for both reading and receiving (not sure if this is what you mean?):

while (swSer.available() > 0)
   SoftwareSerialOut.write(swSer.read());

**Comment by DivanX10** [comment]
@flashmandv 

I don't have a Phillips 3200. The 3200 and 5400 are different machines, different protocols, different controls. Try an ESP32 board. What we mean by noise is that the contact can behave unpredictably and send signals, thereby triggering a command. Read [here](https://randomnerdtutorials.com/esp8266-pinout-reference-gpios/). I won’t tell you about the code - I don’t know.


The pins highlighted in green are OK to use. The ones highlighted in yellow are OK to use, but you need to pay attention because they may have unexpected behavior mainly at boot. The pins highlighted in red are not recommended to use as inputs or outputs.
![image](https://github.com/chris7topher/SmartPhilips2200/assets/64090632/e32a6d6a-b48a-4615-bf9d-c73f29a90d0e)

**Comment by flashmandv** [comment]
Thank you @DivanX10 
I am using D5 (which is green OK) to read and writing to the Serial Tx (GPIO 1)..which is yellow for output as I now see.
I'll try tomorrow using other pin for the write but I'm pessimistic as @chris7topher is using it successfully.
I'll also get ESP32 board and see with it. 
Btw could it be the devboard (NodeMCU)? @chris7topher is using the ESP8266 "naked"..so I don't know if the devboard can affect it.
Thanks

**Comment by flashmandv** [comment]
Hello again guys,

Today I bought ESP32 NodeMCU V1 dev board and **IT WORKED!!**
No ghost start commands. I only tested the machine to display UART forwarding (but I hope everything else will work too).
The ESP8266 devboard may not have worked because of the new firmwares I was flashing (tried installint AT and updating everything). It may also be the Serial UART0 that the devboard USB may have used. I see ESP32 has 3 UARTS so it is best option.

Thanks a lot guys!
